dockstore-webservice/src/main/java/io/dockstore/webservice/core/metrics/MetricsDataS3Client.java

```diff
@@ -133,34 +150,22 @@ public String getMetricsDataFileContent(String toolId, String versionName, Strin
         GetObjectRequest request = GetObjectRequest.builder().bucket(bucketName).key(key).build();
         ResponseInputStream<GetObjectResponse> object = s3.getObject(request);
         return IOUtils.toString(object, StandardCharsets.UTF_8);
-
     }
 
     /**
-     * Get a list of MetricsData for a GA4GH tool version
-     * @param toolId The GA4GH Tool ID
-     * @param toolVersionName The GA4GH ToolVersion name
-     * @return A list of MetricsData
-     * @throws UnsupportedEncodingException
+     * Get the metadata for the MetricsData. The metadata is stored as the S3 object's metadata.
+     * @param metricsData
+     * @return
      */
-    public List<MetricsData> getMetricsData(String toolId, String toolVersionName) throws UnsupportedEncodingException {
-        String key = S3ClientHelper.convertToolIdToPartialKey(toolId) + "/" + URLEncoder.encode(toolVersionName, StandardCharsets.UTF_8);
-        ListObjectsV2Request request = ListObjectsV2Request.builder().bucket(bucketName).prefix(key).build();
-        ListObjectsV2Response listObjectsV2Response = s3.listObjectsV2(request);
-        List<S3Object> contents = listObjectsV2Response.contents();
-        return contents.stream().map(s3Object -> {
-            HeadObjectRequest build = HeadObjectRequest.builder().bucket(bucketName).key(s3Object.key()).build();
-            Map<String, String> metadata = s3.headObject(build).metadata();
-            return convertS3ObjectMetadataToMetricsData(metadata);
-        }).toList();
+    public MetricsDataMetadata getMetricsDataMetadata(MetricsData metricsData) {
+        HeadObjectRequest request = HeadObjectRequest.builder().bucket(bucketName).key(metricsData.s3Key()).build();
+        Map<String, String> s3ObjectMetadata = s3.headObject(request).metadata();
+        long owner = Long.parseLong(s3ObjectMetadata.get(ObjectMetadata.OWNER.toString()));
```

Comment:
It looks like we always set the owner, but if it's ever null, this will throw. Maybe that's OK, just flagging it in case.